### DISCUSSION TITLE
What is the ChaosCenter URL in the Chaos Delegate connection supposed to be

### DISCUSSION THREAD

**Discussion by CS8867** [litmuschaos/litmus#3992] [Q&A]
I am a beginner as far as both K8s and Litmus Chaos is concerned. I wish to connect a kubernetes cluster (minikube) on one machine to the chaos center on a different machine. 
I have two Virtual machines (let's call them A and B) and have installed litmus on both of them. Let's say B is my target cluster. Now, when I open the chaos center dashboard on A, I want to see the target cluster of B in the Chaos Delegates section. I have been following the docs at (https://docs.litmuschaos.io/docs/litmusctl/usage-interactive-mode), but don't really understand what the chaos center url is.
Is my understanding of what the chaos delegate is correct? Or am I getting something totally wrong here. Any help is greatly appreciated.

(This is my first question on a tech forum, so please forgive me if it's poorly structured)

**Comment by gdsoumya** [comment]
The chaos center url is basically the frontend url that you use to access the dashboard.

**Reply by CS8867** [reply]
We access the chaos center locally on the machine using the ip address of the node and the port number mapped to the frontend service right? But that is just for local access. How do I access the chaos center of a remote machine?

**Reply by gdsoumya** [reply]
You need to either use the node port configuration or expose it using load balancer. The litmus documentation has steps for both.

**Reply by CS8867** [reply]
Can you provide the link please?

**Reply by CS8867** [reply]
The frontend service is NodePort by default isn't it? Tried accessing it externally via the internet using the url - 
http://{public-ip-of-machine}:{NodePort}, but its not connecting.
I also tried exposing the frontend service using the load balancer. I had to use minikube tunnel first, following which I was able to see the generated external IP. However, it started from 10, meaning it was a private ip address that it could be used within the kubernetes cluster only (minikube in my case).

**Reply by gdsoumya** [reply]
Minikube is not supposed to be used for production please use eks/gke etc. If you want to expose the services properly over public internet. Minikube is good for local testing only.

**Reply by CS8867** [reply]
Ok, so I forgot to mention one thing (really sorry for this), the two machines that I have are part of the same private network.  
Here are the steps I followed so far -
1. Installed minikube(will look into using GKE instead) and litmus chaos on both. 
2. Litmus chaos frontend dashboard is working fine when accessed locally on both machines.
3. Followed these [docs](https://docs.litmuschaos.io/docs/litmusctl/usage-interactive-mode)
4. I want to access the frontend url of Machine B, from Machine A, as I want to add Machine A as a delegate to Machine B. So I need to expose the chaos-litmus-frontend-service through NodePort (which it is by default)
5. However, at the `litmusctl config set-account` step I am getting the TCP timeout error for the "Host endpoint where litmus is installed:"

Just wanted to confirm if the above steps are correct

**Reply by gdsoumya** [reply]
I see, so you hosted the minikube on machine B and you are able to access the frontend using `http://nodeip_B:nodeport` on machine A? If that works I don't see any reason why litmusctl won't be able to install the agent.

**Reply by CS8867** [reply]
Actually, that's the problem. I cannot access the frontend using `http://nodeip_B:nodeport` from machine A. It's giving me the "Unable to connect" error. By nodeip_B, I am assuming you mean the actual IP of the machine B, which in my case would be the private IP address assigned to machine B by the private network. So I tried load balancer using minikube tunnel, but that doesn't help either...

**Reply by gdsoumya** [reply]
can you run `minikube ip` on machine b and see what the ip is and try using that ip on machine a, it might be the case where minikube actually running on a different virtual network and not the private network.

**Reply by CS8867** [reply]
192.168.49.2
It's the same on both machines

**Reply by gdsoumya** [reply]
That just shows it's not running on your private network but on some docker virtual network. If you can make it run on ur private network it would probably work though that's not in scope for litmus.

**Reply by CS8867** [reply]
Yes, I am running minikube using the docker driver. So you mean to say that this docker network is basically decoupled from my private network and hence `http://nodeip_B:nodeport` does not work?

**Reply by gdsoumya** [reply]
Yes that seems like the most probable issue

**Comment by gdsoumya** [comment]
Closing discussion as the issue is with minikube networking setup and not related to litmus directly.
